oak-run/src/main/java/org/apache/jackrabbit/oak/index/merge/IndexDiff.java

```diff
@@ -0,0 +1,434 @@
+        for(String k : indexes) {
+            if (!k.startsWith("/oak:index/")) {
+                continue;
+            }
+            String nodeName = k.substring("/oak:index/".length());
+            IndexName indexName = IndexName.parse(nodeName);
+            String baseName = indexName.getBaseName();
+            IndexName old = latest.get(baseName);
+            if (old == null) {
+                latest.put(baseName, indexName);
+            } else {
+                if (old.compareTo(indexName) < 0) {
+                    if (old.getCustomerVersion() > 0) {
+                        indexDefinitions.getChildren().remove("/oak:index/" + old.getNodeName());
+                    }
+                    latest.put(baseName, indexName);
+                } else {
+                    indexDefinitions.getChildren().remove("/oak:index/" + nodeName);
+                }
+            }
+        }
+    }
+
+    private static void removeUninterestingIndexProperties(JsonObject indexDefinitions) {
+        for(String k : indexDefinitions.getChildren().keySet()) {
+            JsonObject indexDef = indexDefinitions.getChildren().get(k);
+            indexDef.getProperties().remove("reindexCount");
+            indexDef.getProperties().remove("reindex");
+            indexDef.getProperties().remove("seed");
+            indexDef.getProperties().remove(":version");
+        }
+    }
+
+    private static void simplifyForDisplay(JsonObject json) {
+        for(String k : json.getChildren().keySet()) {
+            JsonObject child = json.getChildren().get(k);
+            simplifyForDisplay(child);
+            child.getProperties().remove("jcr:primaryType");
+            child.getProperties().remove("jcr:created");
+            child.getProperties().remove("jcr:createdBy");
+            child.getProperties().remove("jcr:lastModified");
+            child.getProperties().remove("jcr:uuid");
+            for(String p : child.getProperties().keySet()) {
+                String v = child.getProperties().get(p);
+                if (v.startsWith("\"str:") || v.startsWith("\"nam:")) {
+                    v = "\"" + v.substring(5);
+                    child.getProperties().put(p, v);
+                } else if (v.startsWith("[") && v.contains("\"nam:")) {
+                    v = v.replaceAll("\\[\"nam:", "\\[\"");
+                    v = v.replaceAll(", \"nam:", ", \"");
+                    child.getProperties().put(p, v);
+                } else if (v.startsWith("\":blobId:")) {
+                    String base64 = v.substring(9, v.length() - 1);
+                    try {
+                        String clear = new String(java.util.Base64.getDecoder().decode(base64), "UTF-8");
+                        v = JsopBuilder.encode(clear);
+                        child.getProperties().put(p, v);
+                    } catch (UnsupportedEncodingException e) {
+                        throw new IllegalArgumentException(e);
+                    }
```

Comment:
`StandardCharsets.UTF_8` can be used. No need to try-catch `UnsupportedEncodingException `

```suggestion
                    String clear = new String(java.util.Base64.getDecoder().decode(base64), StandardCharsets.UTF_8);
                    v = JsopBuilder.encode(clear);
                    child.getProperties().put(p, v);
```